### PR TITLE
Remove WorkspaceCollection cmdlets and references

### PR DESCRIPTION
**Pull request by ayeshurun**
## Description

Workspace collection is deprecated. This PR introduces the removal of the related cmdlets and the docs referencing the cmdlets.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Review by laurent-mic — APPROVED** [review]

**Comment by dingmeng-xue** [comment]
@ayeshurun , please update your branch. there are conflicts.

**Comment by ayeshurun** [comment]
@dingmeng-xue conflicts are now resolved

**Comment by ayeshurun** [comment]
@dingmeng-xue @VeryEarly I'd appreciate a review for this PR

**Review by BethanyZhou — COMMENTED** [review]
Is that necessary to rename folder from PowerBIEmbedde* to PowerBI*? Seems it affects our pipeline to detect breaking changes.

**Comment by ayeshurun** [comment]
> Is that necessary to rename folder from PowerBIEmbedde* to PowerBI*? Seems it affects our pipeline to detect breaking changes.

Hi @BethanyZhou, I didn’t rename it, just removed the entire module - folder and files. The src/PowerBIEmbedded folder includes both PowerBI and PowerBIEmbedded folders, and the PowerBIEmbedded inner folder was removed in this PR

File: src/PowerBIEmbedded/PowerBI/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Removed workspace collection cmdlets
```
Comment (isra-fel) [inline src/PowerBIEmbedded/PowerBI/ChangeLog.md:21]: Hey @ayeshurun is there an announcement can be link here?

Comment (isra-fel) [reply src/PowerBIEmbedded/PowerBI/ChangeLog.md:21]: Also, when exactly will the function be deprecated? By "deprecated" do you mean they will stop working or not?

We still see quite some usage of the cmdlets, so we can't just remove the cmdlets while they still work.

Comment (ayeshurun) [reply src/PowerBIEmbedded/PowerBI/ChangeLog.md:21]: @isra-fel The intention is to remove them, as this feature is not available anymore. We shouldn't see any usage.

Comment (msJinLei) [reply src/PowerBIEmbedded/PowerBI/ChangeLog.md:21]: > @isra-fel The intention is to remove them, as this feature is not available anymore. We shouldn't see any usage.

If the customer call the cmdlets that are removed in your PR, what will happen for now?

**Comment by msJinLei** [comment]
@ayeshurun could you rebase your change above the latest main branch?

**Comment by ayeshurun** [comment]
We deferred this change and will ship a PR for cmdlets removal in the next braking change window.

**Pull request closed without merging**